test/integration/dialects/postgres/query-interface.test.js

```diff
@@ -1,180 +1,226 @@
+          // suppress errors here. if create_job doesn't exist thats ok.
+          .reflect();
+      });
+
+      it('creates a stored procedure', function() {
+        const body = 'return test;';
+        const options = {};
+
+        // make our call to create a function
+        return this.queryInterface.createFunction('create_job', [{type:'varchar', name:'test'}], 'varchar', 'plpgsql', body, options)
+          // validate
+          .then(() => this.sequelize.query('select create_job(\'test\');', { type: this.sequelize.QueryTypes.SELECT }))
+          .then(res => {
+            expect(res[0].create_job).to.be.eql('test');
+          });
+      });
+
+      it('treats options as optional', function() {
+        const body = 'return test;';
+
+        // run with null options parameter
+        return this.queryInterface.createFunction('create_job', [{type:'varchar', name:'test'}], 'varchar', 'plpgsql', body, null)
+          // validate
+          .then(() => this.sequelize.query('select create_job(\'test\');', { type: this.sequelize.QueryTypes.SELECT }))
+          .then(res => {
+            expect(res[0].create_job).to.be.eql('test');
+          });
       });
 
-      it('produces an error when missing expected parameters', function () {
-        var body = 'return 1;';
-        var options = {};
-        
+      it('produces an error when missing expected parameters', function() {
+        const body = 'return 1;';
+        const options = {};
+
         return Promise.all([
-          //requires functionName  
-          expect( () => {
+          // requires functionName
+          expect(() => {
             return this.queryInterface.createFunction(null, [{name:'test'}], 'integer', 'plpgsql', body, options);
-          }).to.throw(/createFunction missing some parameters. Did you pass functionName, returnType, language and body/)
-        
-          //requires Parameters array
-          ,expect( () => {
-            return this.queryInterface.createFunction('create_job',null, 'integer', 'plpgsql', body, options);
-          }).to.throw(/function parameters array required/)
-          
-          //requires returnType
-          ,expect( () => {
-            return this.queryInterface.createFunction('create_job', [{type:'varchar',name:'test'}], null, 'plpgsql', body, options);
-          }).to.throw(/createFunction missing some parameters. Did you pass functionName, returnType, language and body/)
-        
-          //requires type in parameter array
-          ,expect( () => {
+          }).to.throw(/createFunction missing some parameters. Did you pass functionName, returnType, language and body/),
+
+          // requires Parameters array
+          expect(() => {
+            return this.queryInterface.createFunction('create_job', null, 'integer', 'plpgsql', body, options);
+          }).to.throw(/function parameters array required/),
```

Comment:
Do these functions really throw synchronous exceptions if wrong parameters are passed (instead of a rejected promise)?